lib/src/command/barback.dart

```diff
@@ -7,13 +7,20 @@ import 'dart:async';
 import 'package:barback/barback.dart';
 import 'package:path/path.dart' as path;
 
+import '../barback/compiler_mode.dart';
 import '../command.dart';
 import '../io.dart';
 import '../log.dart' as log;
 import '../utils.dart';
 
 final _arrow = getSpecial('\u2192', '=>');
 
+final _compilerArgToMode = const {
+  'dart2js': CompilerMode.dart2Js,
+  'dartdevc': CompilerMode.devCompiler,
+  'none': CompilerMode.none,
+};
```

Comment:
Can you turn CompilerMode into a class (enums are almost always pointless in Dart...) and put the logic to loop up a mode by name there?